crates/backend/src/scheduler.rs

```diff
@@ -0,0 +1,93 @@
+use chrono::{DateTime, Local, NaiveDateTime, NaiveTime, TimeZone, Utc};
+use dbus::notification::ScheduledNotification;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+pub struct Scheduler {
+    queue: BinaryHeap<Reverse<ScheduledNotification>>,
+}
+
+impl Scheduler {
+    pub fn new() -> Self {
+        Scheduler {
+            queue: BinaryHeap::new(),
+        }
+    }
+    pub fn add(&mut self, notification: ScheduledNotification) {
+        match Self::parse_time(&notification.time) {
+            Ok(parsed_time) => {
+                dbg!(&parsed_time);
+
+                let scheduled_notification = ScheduledNotification {
+                    time: parsed_time.to_rfc3339(),
+                    data: notification.data,
+                    id: notification.id,
+                };
+                self.queue.push(Reverse(scheduled_notification));
+            }
+            Err(e) => {
+                eprintln!("Error parsing time '{}': {}", notification.time, e);
+            }
+        }
+    }
+
+    fn parse_time(time_str: &str) -> Result<DateTime<Utc>, chrono::ParseError> {
+        let now = Utc::now();
+
+        if let Ok(duration) = humantime::parse_duration(time_str) {
+            return Ok(now + chrono::Duration::from_std(duration).unwrap());
+        }
+
+        let datetime_formats = [
+            "%d.%m.%Y %H:%M",    // 01.01.2025 00:00
+            "%Y-%m-%d %H:%M",    // 01-01-2025 00:00
+            "%d.%m.%Y %I:%M %p", // 01.01.2025 12:00 AM
+            "%Y-%m-%d %I:%M %p", // 01-01-2025 12:00 AM
+        ];
+
+        for format in &datetime_formats {
+            if let Ok(datetime) = NaiveDateTime::parse_from_str(time_str, format) {
+                return Ok(Local
+                    .from_local_datetime(&datetime)
+                    .single()
+                    .expect("Ambiguous local time")
+                    .with_timezone(&Utc));
+            }
+        }
+
+        let time_formats = [
+            "%H:%M",    // 18:45
+            "%I:%M %p", // 06:45 PM
+        ];
+
+        for format in &time_formats {
+            if let Ok(time) = NaiveTime::parse_from_str(time_str, format) {
+                let today = Local::now().date_naive();
+                let datetime = today.and_time(time);
+                return Ok(Local
+                    .from_local_datetime(&datetime)
+                    .single()
+                    .expect("Ambiguous local time")
+                    .with_timezone(&Utc));
```

Comment:
Duplicate code. You can use small function or closure which accepts `&datetime` and returns `chrono::DateTime<Utc>`. The function should be named `from_local_to_utc`.

Also in `expcet` should be correct message. For example, "Local time should have correct format". It's because we _expects_ what should be here.